app/models/pillar.rb

```diff
@@ -104,6 +106,7 @@ def set_pillar(key:, pillar_key:, value:, required_pillars:, errors:)
           errors << "'#{key}' could not be saved#{exp}."
         end
       end
+      Rails.cache.delete "pillars/#{key}"
```

Comment:
i'd put this in an `ensure` block just to be sure that it is always invalidated, even when this method throws an exception for whatever reason

Reply:
Good catch! Updated.